include/lsst/afw/image/detail/StorableMap.h

```diff
@@ -221,6 +221,63 @@ class StorableMap final {
      */
     void clear() noexcept;
 
+    /**
+     * Insert an element into the map, if the map doesn't already contain a
+     * mapping with the same or a conflicting key.
+     *
+     * @tparam T the type of value to insert
+     * @param key key to insert
+     * @param value value to insert
+     *
+     * @return `true` if the insertion took place, `false` otherwise
+     *
+     * @exceptsafe Provides strong exception safety.
+     *
+     * @note It is possible for a key with a value type other than `T` to
+     *       prevent insertion. Callers can safely assume
+     *       `this->contains(key.getId())` as a postcondition, but not
+     *       `this->contains(key)`.
+     */
+    template <typename T, typename std::enable_if_t<std::is_base_of<Storable, T>::value, int> = 0>
+    bool insert(Key<std::shared_ptr<T>> const& key, std::shared_ptr<T> const& value) {
+        static_assert(std::is_const<T>::value,
+                      "Due to implementation constraints, pointers to non-const are not supported.");
+        if (this->contains(key.getId())) {
+            return false;
+        }
+
+        return _contents.emplace(key.getId(), value).second;
+    }
+
+    /**
+     * Insert an element into the map, if the map doesn't already contain a
+     * mapping with a conflicting key.
+     *
+     * @tparam T the type of value to insert (will not compile unless shared
+     *           pointer to a subclass of `Storable`)
+     * @param key key to insert
+     * @param value value to insert
+     *
+     * @return a pair consisting of a strongly-typed key for the value and a
+     *         flag that is `true` if the insertion took place and
+     *         `false` otherwise
+     *
+     * @exceptsafe Provides strong exception safety.
+     *
+     * @warning the type of the compiler-generated key may be surprising.
+     *          Callers should save the returned key if they wish to retrieve
+     *          the value later.
+     */
+    template <typename T>
+    std::pair<Key<T>, bool> insert(std::string const& key, T const& value) {
+        auto strongKey = typehandling::makeKey<T>(key);
+        // Construct return value in advance, so that exception from
+        // copying/moving Key is atomic
```

Comment:
Did this comment need to be broken across the line? Also end in a period.

Reply:
I try to break comments at 80 characters for readability. Including indents, this one would have been 98.